app/utilities/schema.py

```diff
@@ -27,34 +29,35 @@
 }
 
 
-def get_schema_path_map_for_language(language_code):
-    schema_files = []
-
-    for schema_dir in DEFAULT_SCHEMA_DIRS:
-        schema_files.extend(glob(f"{schema_dir}/{language_code}/*.json"))
+def get_schema_path_map_for_language(
+    language_code: str, dirs: Optional[Tuple[str, ...]] = None
+) -> Mapping:
 
     return {
         Path(schema_file).with_suffix("").name: schema_file
-        for schema_file in schema_files
+        for schema_dir in dirs or DEFAULT_SCHEMA_DIRS
+        for schema_file in glob(f"{schema_dir}/{language_code}/*.json")
     }
 
 
-def get_schema_path_map():
-    language_map_codes = ["en", "cy", "ga", "eo"]
+@lru_cache(maxsize=None)
+def get_schema_path_map(dirs: Optional[Tuple[str, ...]] = None) -> Mapping:
+    language_map_codes = {"en", "cy", "ga", "eo"}
 
     return {
-        language_code: get_schema_path_map_for_language(language_code)
+        language_code: get_schema_path_map_for_language(
+            language_code, dirs or DEFAULT_SCHEMA_DIRS
+        )
         for language_code in language_map_codes
     }
 
 
-SCHEMA_PATH_MAP = get_schema_path_map()
-
-
-def schema_exists(language_code, schema_name):
+@lru_cache(maxsize=None)
```

Comment:
Technically it can be called more than one for the same arg if a request is made to a schema for a non default language which doesn't exist and it falls back to the cached English version. However, given that shouldn't really happen, I have removed it.